Added code flow diagram for onboarding

itzg opened the pull request:
# What

I'm not sure if this is the right place to keep this diagram long term, but I wanted to share the notes/diagram that I created to pick apart the primary flow of ingestion code from Kafka to InfluxDB.

FYI switching to "rich diff" show the actual diagram image:
![image](https://user-images.githubusercontent.com/988985/81119503-830a3180-8ef0-11ea-97f5-cb1a710f3b5c.png)

Commit:
Added code flow diagram for onboarding

GeorgeJahad commented:
Another great diagram.  A couple of comments:

1.  What does the word "extends" mean below "LineProtocolBackupService.writeToBackup"?

2.  It looks like the blocks to the lower right refer to the routing service.  The diagram sort of makes that look like it operates in parallel to the influx write, when in fact, iiuc, the influx write depends on it.  Is it worth reorganizing to make that clearer?

Commit:
Clarified the getDbInfo call flow

itzg commented:
> 1. What does the word "extends" mean below "LineProtocolBackupService.writeToBackup"?

Yeah, that was a bit confusing when I denoted it in the first place. Since there's only one implementation of the `LineProtocolBackupService` interface, I just removed it for clarity. It turns out `GCLineProtocolBackupService` uses conditional versions of `Storage` to do in-memory vs real GCP blob storage.

> 2. It looks like the blocks to the lower right refer to the routing service.  The diagram sort of makes that look like it operates in parallel to the influx write, when in fact, iiuc, the influx write depends on it.  Is it worth reorganizing to make that clearer?

Good point. I shifted the InfluxDB write further down to emphasize that's the ultimate goal and added some "return" lines from the routing service branch to show the result feeds into `ingestToInfluxDb`
